.github/workflows/build.yml

```diff
@@ -0,0 +1,346 @@
+# This is the main build pipeline that verifies and publishes the software
+name: Build
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push events
+  push:
+    branches:
+      - main
+      - develop
+      - 'release/**'
+      - 'feature/**'
+      - 'issue/**'
+      - 'issues/**'
+      - 'dependabot/**'
```

Comment:
Should these branches match our naming convention here: https://docs.google.com/document/d/17q0swFmw-oYkkoJ9Bl-IGjY9F6lMIlt7iDiymNcDWEk/edit#heading=h.4y5smdusbxci
This would add the `bug/` and `hotfix/` branches

Reply:
added bug and hotfix patterns